e2e/nomostest/gitproviders/repository.go

```diff
@@ -382,219 +437,220 @@ func (g *Repository) GetAll(dirPath string, recursive bool) []client.Object {
-	g.Git("add", absDestPath)
+	_, err := g.Git("add", absDestPath)
+	return err
 }
 
 // Remove deletes `file` from the git repository.
 // If `file` is a directory, deletes the directory.
 // Returns error if the file does not exist.
 // Does not commit/push.
-func (g *Repository) Remove(path string) {
-	g.T.Helper()
-
+func (g *Repository) Remove(path string) error {
 	absPath := filepath.Join(g.Root, path)
 
-	err := os.RemoveAll(absPath)
-	if err != nil {
-		g.T.Fatal(err)
+	if err := os.RemoveAll(absPath); err != nil {
+		return errors.Wrapf(err, "deleting path: %s", absPath)
 	}
 
-	g.Git("add", absPath)
+	_, err := g.Git("add", absPath)
+	return err
 }
 
 // Exists returns true if the file or directory exists at the specified path.
-func (g *Repository) Exists(path string) bool {
-	g.T.Helper()
-
+func (g *Repository) Exists(path string) (bool, error) {
 	absPath := filepath.Join(g.Root, path)
 
 	_, err := os.Stat(absPath)
 	if err != nil {
 		if os.IsNotExist(err) {
-			return false
+			return false, nil
 		}
-		g.T.Fatal(err)
+		return false, errors.Wrapf(err, "describing file path: %s", absPath)
 	}
-	return true
+	return true, nil
 }
 
 // CommitAndPush commits any changes to the git repository, and
 // pushes them to the git server.
 // We don't care about differentiating between committing and pushing
 // for tests.
-func (g *Repository) CommitAndPush(msg string) {
-	g.T.Helper()
-	g.CommitAndPushBranch(msg, MainBranch)
+func (g *Repository) CommitAndPush(msg string) error {
+	return g.CommitAndPushBranch(msg, MainBranch)
 }
 
 // CommitAndPushBranch commits any changes to the git branch, and
 // pushes them to the git server.
-func (g *Repository) CommitAndPushBranch(msg, branch string) {
-	g.T.Helper()
-
-	g.Git("commit", "-m", msg)
-
-	g.Logger.Infof("[repo %s] committing %q (%s)", path.Base(g.Root), msg, g.Hash())
-	g.Push(branch, "-f")
+func (g *Repository) CommitAndPushBranch(msg, branch string) error {
+	g.Logger.Infof("[repo %s] committing: %s", path.Base(g.Root), msg)
+	if _, err := g.Git("commit", "-m", msg); err != nil {
+		return err
+	}
+	commit, err := g.Hash()
+	if err != nil {
+		return err
+	}
+	g.Logger.Infof("[repo %s] pushing commit: %s", path.Base(g.Root), commit)
+	return g.Push(branch, "-f")
 }
 
 // Push pushes the provided refspec to the git server.
 // Performs a retry using RemoteURL, which may change if the port forwarding restarts.
-func (g *Repository) Push(refspec string, flags ...string) {
-	var out []byte
+func (g *Repository) Push(args ...string) error {
 	took, err := retry.Retry(1*time.Minute, func() error {
-		args := []string{"push"}
-		args = append(args, flags...)
 		remoteURL, err := g.GitProvider.RemoteURL(g.RemoteRepoName)
 		if err != nil {
 			return err
 		}
-		args = append(args, remoteURL, refspec)
-		cmd := g.gitCmd(args...)
-		out, err = cmd.CombinedOutput()
+		_, err = g.Git(append([]string{"push", remoteURL}, args...)...)
 		return err
 	})
 	if err != nil {
-		g.T.Log(string(out))
```

Comment:
Was it intentional to remove the stdout/stderr logging?

Reply:
The `Git` method handles logging on error now, so the callers don't need to.